flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -501,6 +537,11 @@ public void alterTable(CatalogBaseTable table, ObjectIdentifier objectIdentifier
 	 *                          does not exist.
 	 */
 	public void dropTable(ObjectIdentifier objectIdentifier, boolean ignoreIfNotExists) {
+		if (temporaryTables.containsKey(objectIdentifier)) {
```

Comment:
this behavior feels a bit abnormal to me. Why cannot users drop a catalog table when a temp table with same name space exist?

From a sql perspective, users should be able to run "DROP TABLE" and "DROP TEMP TABLE" independently

Reply:
Just curious about this, I took a look to spark:
```
   * If a database is specified in `name`, this will drop the table from that database.
   * If no database is specified, this will first attempt to drop a temporary view with
   * the same name, then, if that does not exist, drop the table from the current database.
```
Curious about other databases, I don't know how they behave.

Reply:
Spark does not have 3-part qualified temporary objects. We followed MySQL on that topic.

This behavior is introduced to prevent unintentional drops of temporary tables, which might have very costly undesirable effects. Users are usually accustomed with `DROP TABLE`, but not necessarily with `DROP TEMPORARY TABLE`. If a user issues a `DROP TABLE` with the intent of dropping the temporary table it will actually drop in a drop of the permanent one.

Dropping tables is a very invasive operation. I tried to design the behaviour in a way to force the user make a really conscious decision on which table should be dropped.

BTW, as a side note, this behaviour is described in the FLIP-64. There were no objections towards that behaviour during the discussion.